File: SourceAFIS/General/RangeF.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SourceAFIS.General
{
    public struct RangeF
    {
        public double Begin;
        public double End;

        public double Length { get { return End - Begin; } }

        public RangeF(double begin, double end)
        {
            Begin = begin;
            End = end;
        }

        public double GetFraction(double value)
        {
            return (value - Begin) / Length;
        }

        public double Interpolate(double fraction)
        {
            return Calc.Interpolate(Begin, End, fraction);
        }
    }
}

File: java/sourceafis/src/sourceafis/general/RangeF.java
/**
 * @author Veaceslav Dubenco
 * @since 10.10.2012
 */
package sourceafis.general;

/**
 * 
 */
public class RangeF {
	public float Begin;
	public float End;

	public float getLength() {
		return End - Begin;
	}

	public RangeF(float begin, float end) {
		Begin = begin;
		End = end;
	}

	public float GetFraction(float value) {
		return (value - Begin) / getLength();
	}

	public float Interpolate(float fraction) {
		return Calc.Interpolate(Begin, End, fraction);
	}

}
